src/vanilla.ts

```diff
@@ -10,6 +10,11 @@ export type StateListener<T> = (state: T, previousState: T) => void
 export type StateSliceListener<T> = (slice: T, previousSlice: T) => void
 export interface Subscribe<T extends State> {
   (listener: StateListener<T>): () => void
+  (
+    listener: StateSliceListener<any>,
+    selector?: StateSelector<T, any>,
+    equalityFn?: EqualityChecker<any>
+  ): () => void
```

Comment:
You seem to get the point. (I was replying to your old comment, and deleted.)

```suggestion
  (
    listener: StateSliceListener<T>,
    selector: undefined,
    equalityFn?: EqualityChecker<T>
  ): () => void
```

Does this work?

Reply:
It does not sadly:

```
Argument of type '((state: unknown) => AnyObject) | undefined' is not assignable to parameter of type 'StateSelector<S, unknown>'.
      Type 'undefined' is not assignable to type 'StateSelector<S, unknown>
```

Reply:
My fix works, and I don't really see a downside to including it. Zustand gives you two ways to subscribe and preserve types, but if you need to do some dynamic untyped stuff you can also do that. If I want to shoot myself in the foot, Zustand shouldn't stand in my way xD